### PR TITLE
Adds documentation for editor-expand.js

### LINKED ISSUE #8: Document wp-admin/js/editor-expand.js

### PR DESCRIPTION
**Pull request by terw-dan**
Fixes #8 

Adds documentation to src/wp-admin/js/editor-expand.js

Don't forget to give credits to @IreneStr

### CODE REVIEW

File: src/wp-admin/js/editor-expand.js [hunk 1]
```diff
@@ -6,7 +6,13 @@
 		$adminBar = $( '#wpadminbar' ),
 		$footer = $( '#wpfooter' );
 
-	/* Autoresize editor. */
+	/**
+	 * @summary Handles the resizing of the editor
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: Missing `.`.

File: src/wp-admin/js/editor-expand.js [hunk 2]
```diff
@@ -73,6 +89,18 @@
 			}
 		}, 300 );
 
+		/**
+		 * @summary Resizes the text editor depending on the old text length.
+		 *
+		 * If there is an mceEditor and it is hidden, resizes the editor depending
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: Missing `it` between `[...] is hidden, resizes[...]`.

File: src/wp-admin/js/editor-expand.js [hunk 3]
```diff
@@ -73,6 +89,18 @@
...
+		 * on the old text length. If the current length of the text is smaller than
+		 * the old text length, shrink the text area. Otherwise resize the editor to
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: `[...] it shrinks the textarea. Otherwise, it resizes [...]`.

File: src/wp-admin/js/editor-expand.js [hunk 4]
```diff
@@ -94,6 +122,18 @@
 			oldTextLength = length;
 		}
 
+		/**
+		 * @summary Gets the heights and widths of elements.
+		 *
+		 * Gets the heights of the window, the adminbar, the tools, the menu,
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: Height should be singular.

File: src/wp-admin/js/editor-expand.js [hunk 5]
```diff
@@ -110,50 +150,77 @@
...
-			// Adjust for hidden
+			// Adjust for hidden menubar.
 			if ( heights.menuBarHeight < 3 ) {
 				heights.menuBarHeight = 0;
 			}
 		}
 
 		// We need to wait for TinyMCE to initialize.
+		/**
+		 * @summary Binds all necessary functions for editor expand to the editor,
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: Not entirely sure, but I think the comma is unnecessary.

File: src/wp-admin/js/editor-expand.js [hunk 6]
```diff
@@ -110,50 +150,77 @@
...
+		 * when the editor is initialized.
+		 *
+		 * @since 4.0.0
+		 *
+		 * @param {event} event The TinyMCE editor init event
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: Missing `.`.

File: src/wp-admin/js/editor-expand.js [hunk 7]
```diff
@@ -110,50 +150,77 @@
...
+		 * @param {object} editor The editor to bind the vents on.
+		 *
+		 * @returns
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: Missing return value and description.

File: src/wp-admin/js/editor-expand.js [hunk 8]
```diff
@@ -110,50 +150,77 @@
...
+		 */
 		$document.on( 'tinymce-editor-init.editor-expand', function( event, editor ) {
+			// VK contains the type of key pressed. VK = virtual keyboard.
 			var VK = window.tinymce.util.VK,
+				/**
+				 * @summary Hides any float panel with a hover state, and tooltips.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: I'd consider rewriting this to something like: `Hides any float panel with a hover state. Additionally hides tooltips.`.

File: src/wp-admin/js/editor-expand.js [hunk 9]
```diff
@@ -110,50 +150,77 @@
...
+				 *
+				 * @returns {void}
+				 */
 				hideFloatPanels = _.debounce( function() {
 					! $( '.mce-floatpanel:hover' ).length && window.tinymce.ui.FloatPanel.hideAll();
 					$( '.mce-tooltip' ).hide();
 				}, 1000, true );
 
-			// Make sure it's the main editor.
+			// Makes sure it's the main editor.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: Single line comments adhere the 'Make' form. Also applies to lines 189, 192, 195, 218, 223, 245, 279 and 282.

File: src/wp-admin/js/editor-expand.js [hunk 10]
```diff
@@ -110,50 +150,77 @@
...
+			 * @summary Gets the offset of the editor.
+			 *
+			 * @returns {Number|Boolean} Returns the offset of the editor
+			 * or false if there is no offset height.
+			 */
 			function mceGetCursorOffset() {
 				var node = editor.selection.getNode(),
 					range, view, offset;
 
+				// If editor.wp.getView and the selection node from the editor selection
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: Consider using block-style comments for multi-line comments.

File: src/wp-admin/js/editor-expand.js [hunk 11]
```diff
@@ -162,61 +229,120 @@
...
 
 				mceScroll( key );
 			}
 
+			/**
+			 * @summary Makes sure the cursor is always visible in the editor.
+			 *
+			 * @since 4.1.0
+			 *
+			 * Makes sure the cursor is kept between the toolbars of the editor and scrolls
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: Long description should appear before the `@since` property.

File: src/wp-admin/js/editor-expand.js [hunk 12]
```diff
@@ -162,61 +229,120 @@
...
+			 * others to the top/bottom of the *window* when moving the cursor out of the viewport.
+			 *
+			 * @param {string} key The key code of the pressed key.
+			 */
 			function mceScroll( key ) {
 				var offset = mceGetCursorOffset(),
 					buffer = 50,
 					cursorTop, cursorBottom, editorTop, editorBottom;
 
+				// Doesn't scroll if there is no offset.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: `Don't scroll [...]`

File: src/wp-admin/js/editor-expand.js [hunk 13]
```diff
@@ -162,61 +229,120 @@
...
+
+				// Subtract the buffer from the cursorTop.
 				cursorTop = cursorTop - buffer;
+
+				// Add the buffer to the cursorBottom.
 				cursorBottom = cursorBottom + buffer;
 				editorTop = heights.adminBarHeight + heights.toolsHeight + heights.menuBarHeight + heights.visualTopHeight;
+
+				/*
+				 * Sets the editorBottom based on the window Height, and adds the bottomHeight and statusBarHeight if the
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: `Set [...], and add [...]`.

File: src/wp-admin/js/editor-expand.js [hunk 14]
```diff
@@ -162,61 +229,120 @@
...
+				 * advanced editor is enabled.
+				 */
 				editorBottom = heights.windowHeight - ( advanced ? heights.bottomHeight + heights.statusBarHeight : 0 );
 
-				// Don't scroll if the node is taller than the visible part of the editor
+				// Doesn't scroll if the node is taller than the visible part of the editor.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: `Don't [...]`.

File: src/wp-admin/js/editor-expand.js [hunk 15]
```diff
@@ -162,61 +229,120 @@
...
 				if ( editorBottom - editorTop < offset.height ) {
 					return;
 				}
 
+				/*
+				 * If the cursorTop is smaller than the editorTop and the up, left
+				 * or backspace key is pressed, the editor scrolls to the position defined
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: `[...] scroll the editor to the [...]`.

File: src/wp-admin/js/editor-expand.js [hunk 16]
```diff
@@ -162,61 +229,120 @@
...
+				 * by the cursorTop, pageYOffset and editorTop.
+				 */
 				if ( cursorTop < editorTop && ( key === VK.UP || key === VK.LEFT || key === VK.BACKSPACE ) ) {
 					window.scrollTo( window.pageXOffset, cursorTop + window.pageYOffset - editorTop );
+				/*
+				 * If any other key is pressed or the cursorTop is bigger than the editorTop,
+				 * the editor scrolls to the position defined by the cursorBottom,
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See previous comment.

File: src/wp-admin/js/editor-expand.js [hunk 17]
```diff
@@ -162,61 +229,120 @@
...
+				 * pageYOffset and editorBottom.
+				 */
 				} else if ( cursorBottom > editorBottom ) {
 					window.scrollTo( window.pageXOffset, cursorBottom + window.pageYOffset - editorBottom );
 				}
 			}
 
+			/**
+			 * @summary If the editor is fullscreen, call adjust.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: `[...] calls [...]`.

File: src/wp-admin/js/editor-expand.js [hunk 18]
```diff
@@ -275,15 +435,28 @@
...
+
+				// Adjust "immediately".
 				mceBind();
 				initialResize( adjust );
 			}
 		} );
 
-		// Adjust the toolbars based on the active editor mode.
+		/**
+		 * @summary Adjusts the toolbars heights and positions..
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: Redundant `.`.

File: src/wp-admin/js/editor-expand.js [hunk 19]
```diff
@@ -299,11 +472,15 @@
 				$top, $editor, sidebarTop, footerTop, canPin,
 				topPos, topHeight, editorPos, editorHeight;
 
-			// Refresh the heights
+			/*
+			 * Refreshes the heights if type isn't 'scroll'
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 20]
```diff
@@ -299,11 +472,15 @@
...
+			 * or heights.windowHeight isn't set.
+			 */
 			if ( resize || ! heights.windowHeight ) {
 				getHeights();
 			}
 
+			// Resizes on resize event when the editor is in text mode.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 21]
```diff
@@ -318,7 +495,7 @@
 				topHeight = heights.textTopHeight;
 			}
 
-			// TinyMCE still initializing.
+			// Returns if TinyMCE is still intializing.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 22]
```diff
@@ -357,11 +537,9 @@
 					$bottom.attr( 'style', '' );
 				}
 			} else {
-				// Maybe pin the top.
+				// Checks if the top is not already in a fixed position.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 23]
```diff
@@ -384,9 +562,8 @@
 						top: heights.adminBarHeight + heights.toolsHeight + heights.menuBarHeight,
 						width: contentWrapWidth - ( borderWidth * 2 ) - ( visual ? 0 : ( $top.outerWidth() - $top.width() ) )
 					} );
-				// Maybe unpin the top.
+					// Checks if the top is already in a fixed position.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 24]
```diff
@@ -435,9 +611,9 @@
 					}
 				}
 
-				// Maybe adjust the bottom bar.
+				// Checks if the bottom is not already in a fixed position.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 25]
```diff
@@ -435,9 +611,9 @@
...
 				if ( ( ! fixedBottom || ( resize && advanced ) ) &&
-						// +[n] for the border around the .wp-editor-container.
+						// Adds borderWidth for the border around the .wp-editor-container.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 26]
```diff
@@ -468,26 +644,33 @@
 				}
 			}
 
-			// Sidebar pinning
-			if ( $postboxContainer.width() < 300 && heights.windowWidth > 600 && // sidebar position is changed with @media from CSS, make sure it is on the side
-				$document.height() > ( $sideSortables.height() + postBodyTop + 120 ) && // the sidebar is not the tallest element
-				heights.windowHeight < editorHeight ) { // the editor is taller than the viewport
+			// The postbox container is positioned with @media from CSS. This makes sure it is pinned on the side.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305. `Ensure [...]`.

File: src/wp-admin/js/editor-expand.js [hunk 27]
```diff
@@ -468,26 +644,33 @@
...
+			if ( $postboxContainer.width() < 300 && heights.windowWidth > 600 &&
+
+				// Checks if the sidebar is not taller than the document height.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 28]
```diff
@@ -468,26 +644,33 @@
...
+				$document.height() > ( $sideSortables.height() + postBodyTop + 120 ) &&
+
+				// Checks if the editor is taller than the viewport.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 29]
```diff
@@ -575,38 +759,73 @@
...
+		 *
+		 * @since 4.0.0
+		 *
+		 * @returns {void}
+		 */
 		function on() {
-			// Scroll to the top when triggering this from JS.
-			// Ensures toolbars are pinned properly.
+			/*
+			 * Scrolls to the top when triggering this from JS.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 30]
```diff
@@ -575,38 +759,73 @@
...
+			 * Ensures toolbars are pinned properly.
+			 */
 			if ( window.pageYOffset && window.pageYOffset > pageYOffsetAtTop ) {
 				window.scrollTo( window.pageXOffset, 0 );
 			}
 
 			$wrap.addClass( 'wp-editor-expand' );
 
-			// Adjust when the window is scrolled or resized.
+			// Adjusts when the window is scrolled or resized.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 31]
```diff
@@ -575,38 +759,73 @@
...
 			$window.on( 'scroll.editor-expand resize.editor-expand', function( event ) {
 				adjust( event.type );
 				afterScroll();
 			} );
 
-			// Adjust when collapsing the menu, changing the columns, changing the body class.
+			/*
+		 	 * Adjusts when collapsing the menu, changing the columns
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 32]
```diff
@@ -628,7 +847,7 @@
 			$textEditor.on( 'focus.editor-expand input.editor-expand propertychange.editor-expand', textEditorResize );
 			mceBind();
 
-			// Adjust when entering/exiting fullscreen mode.
+			// Adjusts when entering or exiting fullscreen mode.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 33]
```diff
@@ -658,8 +884,10 @@
 				height = 5000;
 			}
 
-			// Scroll to the top when triggering this from JS.
-			// Ensures toolbars are reset properly.
+			/*
+			 * Scrolls to the top when triggering this from JS.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 34]
```diff
@@ -671,10 +899,10 @@
 			$textEditor.off( '.editor-expand' );
 			mceUnbind();
 
-			// Adjust when entering/exiting fullscreen mode.
+			// Adjusts when entering or exiting fullscreen mode.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 35]
```diff
@@ -671,10 +899,10 @@
...
 			fullscreen && fullscreen.pubsub.unsubscribe( 'hidden', fullscreenHide );
 
-			// Reset all css
+			// Resets all css
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 36]
```diff
@@ -694,18 +922,19 @@
...
+			// If there is a height found in the user setting.
 			if ( height ) {
 				$textEditor.height( height );
 			}
 
 			$document.trigger( 'editor-expand-off' );
 		}
 
-		// Start on load
+		// Starts on load.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 37]
```diff
@@ -714,7 +943,7 @@
 			}
 		}
 
-		// Show the on/off checkbox
+		// Shows the on/off checkbox.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 38]
```diff
@@ -726,14 +955,20 @@
 			}
 		});
 
-		// Expose on() and off()
+		// Exposes on() and off().
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 39]
```diff
@@ -777,12 +1012,26 @@
 			mouseY = event.pageY;
 		} );
 
+		/**
+		 * @summary Recalculates the editor rectangle.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: What rectangle? Consider adding a bit more copy.

File: src/wp-admin/js/editor-expand.js [hunk 40]
```diff
@@ -839,6 +1116,13 @@
 			}
 		}
 
+		/**
+		 * @summary Runs on or off.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: This could use a bit more copy. Perhaps an example?

File: src/wp-admin/js/editor-expand.js [hunk 41]
```diff
@@ -859,12 +1163,13 @@
 				isMac = ( window.navigator.platform.indexOf( 'Mac' ) > -1 );
 			}
 
-			// fadeIn and return on Escape and keyboard shortcut Alt+Shift+W and Ctrl+Opt+W.
+			// Fades in and returns on Escape and keyboard shortcut Alt+Shift+W and Ctrl+Opt+W.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 42]
```diff
@@ -859,12 +1163,13 @@
...
 			if ( key === 27 || ( key === 87 && event.altKey && ( ( ! isMac && event.shiftKey ) || ( isMac && event.ctrlKey ) ) ) ) {
 				fadeIn( event );
 				return;
 			}
 
+			// Returns if any of the following keys or combinations of keys is pressed.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 43]
```diff
@@ -918,7 +1223,7 @@
 
 						$window.off( 'scroll.focus' );
 					} )
-					// Fade in when the mouse moves away form the editor area.
+					// Fades in when the mouse moves away form the editor area.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 44]
```diff
@@ -959,7 +1264,8 @@
 						x = nx;
 						y = ny;
 					} )
-					// When the overlay is touched, always fade in and cancel the event.
+
+					// When the overlay is touched, fades in and cancels the event.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 45]
```diff
@@ -1151,9 +1532,10 @@
 					mceBind();
 				}
 
+				// Binds and unbinds based on the distraction free writing focus.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 46]
```diff
@@ -1151,9 +1532,10 @@
...
 				$document.on( 'dfw-on.focus', mceBind ).on( 'dfw-off.focus', mceUnbind );
 
-				// Make sure the body focuses when clicking outside it.
+				// Focuses the editor when it is the target of the click event.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

File: src/wp-admin/js/editor-expand.js [hunk 47]
```diff
@@ -1162,9 +1544,20 @@
...
+		 *
+		 * @param {event} event The quicktags init event.
+		 * @param {object} editor The editor to bind events on.
+		 *
+		 * @returns {void}
+		 */
 		$document.on( 'quicktags-init', function( event, editor ) {
 			var $button;
 
+			// Binds the distraction free writing events if the distraction free writing button is available.
```
Comment (jcomack) [inline src/wp-admin/js/editor-expand.js]: See comment made on line 305.

**Comment by moorscode** [comment]
Patch submitted https://core.trac.wordpress.org/ticket/41068

**Comment by ntwb** [comment]
Commit: https://core.trac.wordpress.org/changeset/41310

This issue can now be closed

**Pull request closed without merging**